code/player/Player.cs

```diff
@@ -100,15 +117,51 @@ public override void StartTouch(Entity other)
             base.StartTouch(other);
         }
 
-        private void TickInspectPlayerCorpse()
+        private void TickAttemptInspectPlayerCorpse()
         {
-            // TODO: Handle role specific corpse inspecting.
-            if (!Input.Pressed(InputButton.Use))
+            if (IsClient)
             {
-                return;
+                if (InspectMenu.Instance?.IsShowing ?? false)
+                {
+                    // Menu is showing already, bail out.
+                    return;
+                }
             }
 
-            TraceResult trace = Trace.Ray(EyePos, EyePos + EyeRot.Forward * 80f)
+            if (IsServer)
+            {
+                using (Prediction.Off())
+                {
+                    To client = To.Single(this);
+                    PlayerCorpse playerCorpse = IsLookingAtPlayerCorpse();
+                    if (playerCorpse != null)
+                    {
+                        if (Input.Down(InputButton.Use) && !playerCorpse.IsIdentified)
+                        {
+                            playerCorpse.IsIdentified = true;
+                            Client playerCorpseInfo = playerCorpse.Player.GetClientOwner();
+                            ClientDisplayIdentifiedMessage(this.Controller.Client.SteamId,
+                                                        this.Controller.Client.Name,
+                                                        playerCorpseInfo.SteamId,
+                                                        playerCorpseInfo.Name,
+                                                        playerCorpse.Player.Role.ToString());
```

Comment:
```suggestion

                            ClientDisplayIdentifiedMessage(
                                this.Controller.Client.SteamId,
                                this.Controller.Client.Name,
                                playerCorpseInfo.SteamId,
                                playerCorpseInfo.Name,
                                playerCorpse.Player.Role.ToString()
                            );
```

This would match our code style far more, but I don't know about the first bracket (new line or not)